modules/core/src/test/scala/io/janstenpickle/hotswapref/HotswapRefSpec.scala

```diff
@@ -48,11 +48,11 @@ class HotswapRefSpec extends AnyFlatSpec with Matchers with TestInstances {
       )
       _ <- Resource.eval(IO.sleep(1.second)) // give it a chance to update
       _ <- Resource.eval(hotswap.swap(Resource.pure(ref1)).start)
-      _ <- Resource.eval(IO.sleep(1.second)) // give it a chance to update
+      _ <- Resource.eval(IO.sleep(1.second)) // give it a chance to allocate a new resource and refresh the holder ref
```

Comment:
I think we should rework this test, it's really hard to follow. Can you please walk me through what this test aims to achieve?